syft/pkg/cataloger/dart/parse_pubspec_lock.go

```diff
@@ -68,8 +70,28 @@ func parsePubspecLock(_ context.Context, _ file.Resolver, _ *generic.Environment
 	}
 
 	var names []string
-	for name := range p.Packages {
+	for name, pkg := range p.Packages {
 		names = append(names, name)
+
+		if pkg.Source == "sdk" && pkg.Version == "0.0.0" {
+			// Packages that are delivered as part of an SDK (e.g. Flutter) have their
+			// version set to "0.0.0" in the package definition. The actual version
+			// should refer to the SDK version, which is defined in a dedicated section
+			// in the pubspec.lock file and uses a version range constraint.
+			//
+			// If such a package is detected, look up the version range constraint of
+			// its matching SDK, and set the minimum supported version as its new version.
+			sdkName := pkg.Description.Name
+			sdkVersion, err := p.getSdkVersion(sdkName)
+
+			if err != nil {
```

Comment:
if we run into an error, I think we should clear the version from the package entirely (since 0.0.0 is not correct). This would allow the package name/version compliance policy to be correctly applied.

Reply:
Hope you understood you correct with this: if resolving the version fails, the package itself is now skipped and not added to the package list.